### PR TITLE
Introducing @ScheduledTask annotation for vender independent task scheduling

### PR DESCRIPTION
**Pull request by tgeens**
Dynamic Extensions has support for scheduling quartz jobs since DE 1.3 with the `@ScheduledQuartzJob` annotation. Alfresco 6.x bumped the dependency on Quartz from 1.x to 2.x, which comes with a few breaking API changes. In most cases, this does not break user-code.

To decouple the DE API from Quartz, this PR abstracts away the direct dependency on Quartz (and the breaking API changes between Quartz 1.x and 2.x).

`@ScheduledTask` replaces the now deprecated `@ScheduledQuartzJob` in DE 2.0 + there is a simple `eu.xenit.dynamicextensionsalfresco.schedule.Task` interface that your scheduled job/task should implement.

### CODE REVIEW

File: annotations-runtime-schedule/scheduler-quartz-1/src/main/java/com/github/dynamicextensionsalfresco/schedule/quartz/ClusterLockedQuartzJob.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package com.github.dynamicextensionsalfresco.schedule.quartz;
+
+import org.alfresco.schedule.AbstractScheduledLockedJob;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+
+public class ClusterLockedQuartzJob extends AbstractScheduledLockedJob {
```
Comment (kerkhofsd) [inline annotations-runtime-schedule/scheduler-quartz-1/src/main/java/com/github/dynamicextensionsalfresco/schedule/quartz/ClusterLockedQuartzJob.java]: Identical in the quartz-1 and quartz-2 subproject, can this be moved to the `:annotations-runtime-schedule`?

File: annotations-runtime/build.gradle [hunk 2]
```diff
@@ -9,19 +9,19 @@ subprojects {
...
         compile project(':annotations')
         compile project(":webscripts")
+        compile project(':annotations-runtime-schedule') // schedule abstraction
 
         compileOnly "org.apache.felix:org.apache.felix.framework:${project.ext.felixVersion}"
 
         compileOnly "org.slf4j:slf4j-api"
         compileOnly("org.alfresco.surf:spring-webscripts") { 
-            exclude group: "org.quartz-scheduler"
+            exclude group: "org.schedule-schedule"
```
Comment (kerkhofsd) [inline annotations-runtime/build.gradle]: Is this correct?

**Comment by kerkhofsd** [comment]
Integration tests are failing, can you have a look?

File: annotations-runtime/build.gradle (hunk 2 above)
Comment (tgeens) [reply annotations-runtime/build.gradle]: Good catch, that's some IntelliJ refactoring trying to be helpful :-)

File: annotations-runtime-schedule/scheduler-quartz-1/src/main/java/com/github/dynamicextensionsalfresco/schedule/quartz/ClusterLockedQuartzJob.java (hunk 1 above)
Comment (tgeens) [reply annotations-runtime-schedule/scheduler-quartz-1/src/main/java/com/github/dynamicextensionsalfresco/schedule/quartz/ClusterLockedQuartzJob.java]: I was intentionally keeping `:annotations-runtime-schedule` free from any quartz or Alfresco dependency.

Comment (kerkhofsd) [reply annotations-runtime-schedule/scheduler-quartz-1/src/main/java/com/github/dynamicextensionsalfresco/schedule/quartz/ClusterLockedQuartzJob.java]: Agreed

**Pull request merged**